sdk/python/packages/flet-core/src/flet_core/page.py

```diff
@@ -974,42 +1001,112 @@ def __on_invoke_method_result(self, e):
         self.__method_call_results[evt] = (result.result, result.error)
         evt.set()
 
+    # Open / Close
+    def __open_control(
+        self, control_name: str, control: Control, base: "Union[Page, Offstage]" = None
+    ) -> None:
+        if not base:
+            base = self
+
+        if control:
+            control.open = True
+            base.update()
+        else:
+            raise Exception(
+                f"Please define {control_name} before show_{control_name}() method"
+            )
+
+    def __close_control(
+        self, control_name: str, control: Control, base: "Union[Page, Offstage]" = None
+    ) -> None:
+        if not base:
+            base = self
+
+        if control:
+            control.open = False
+            base.update()
+        else:
+            raise Exception(
+                f"Please define {control_name} before close_{control_name}() method"
+            )
+
     #
     # SnackBar
     #
-    def show_snack_bar(self, snack_bar: SnackBar):
-        self.__offstage.snack_bar = snack_bar
-        self.__offstage.snack_bar.open = True
-        self.__offstage.update()
+    @overload
+    def open_snack_bar(self) -> None:
+        ...
```

Comment:
Why do we need guys like these?

Reply:
Btw, I was thinking about unifying of all "openable" controls and having a single `open()` method for everything. Like adding `OpenableControl` (what would be a better name? :) base class with at least `open` property and then having a list of such controls in `__offstage`. Does it make sense?

Reply:
I can open PR after merging this with realisation of open() method